src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -440,7 +434,36 @@ class ExtractFunctionNameSegment(BaseSegment):
     """
 
     type = "function_name"
-    match_grammar: Matchable = Ref("ExtractFunctionName")
+    match_grammar: Matchable = StringParser(
+        "EXTRACT",
+        CodeSegment,
+        name="function_name_identifier",
+        type="function_name_identifier",
+    )
+
+
+class NormalizeFunctionNameSegment(BaseSegment):
+    """NORMALIZE function name segment.
+
+    Need to be able to specify this as type function_name
+    so that linting rules identify it properly
+    """
+
+    type = "function_name"
+    match_grammar: Matchable = OneOf(
+        StringParser(
```

Comment:
This seems repetitive, two `StringParser`s with only the first parameter varying. Worth trying to combine these somehow?

Reply:
Any ideas how, other than using `RegexParser` that @WittierDinosaur didn't like (that's what I originally had).

Reply:
I don't suppose `StringParser` would accept a `OneOf()` argument rather than just a string?

This is just a nitpick, so feel free to ignore.

Reply:
Nah it doesn't accept that. It's a Segment builder, rather than something able to use segments itself.

Will merge as is.